__tests__/endpoints.test.js

```diff
@@ -0,0 +1,395 @@
+const request = require('supertest')
+const app = require('../app.js')
+const db = require('../db/data/test-data/index.js')
+const seed = require('../db/seeds/seed.js')
+
+beforeEach(()=> seed(db))
+
+describe('Topic Tests', () => {
+    test('Status 200: GETs all Topics',()=>{
+        return request(app)
+        .get('/api/topics')
+        .expect(200)
+        .then(({body })=>{
+            const topics = body
+            expect(topics).toHaveLength(3)
+            topics.forEach((topic)=> {
+                expect(topic).toMatchObject({
+                    description : expect.any(String),
+                    slug : expect.any(String),
+            })
+
+            })
+        })
+    })
+})
+
+describe('Endpoint Tests', () => {
+    test('Status 200: GETs all EndPoints Info',()=>{
+        return request(app)
+        .get('/api')
+        .expect(200)
+        .then((data)=>{
+            const endpoints = data.body
+            
+            for (let endpoint in endpoints) {
+             const endpointObject = endpoints[endpoint]
+             const length = Object.keys(endpointObject).length
+
+             switch(length) {
+                case 1:
+                  // code block
+                  expect(endpointObject).toMatchObject({
+                    description : expect.any(String),
+                })
+                  break;
+                case 2:
+                  // code block
+                  expect(endpointObject).toMatchObject({
+                    description : expect.any(String),
+                    queries : expect.any(Array),
+                })
+                  break;
+                case 3:
+                    expect(endpointObject).toMatchObject({
+                        description : expect.any(String),
+                        queries : expect.any(Array),
+                        exampleResponse : expect.any(Object),
+                    })
+                  break;
+                default:
+              }
+
+
+            }
+        })
+    })
+})
+
+    describe('Article Tests', () => {
+    describe('For retrieving an individual article', () => {
+    test('Status 200: GETs an Article when given an article ID which we have',()=>{
+        return request(app)
+        .get('/api/articles/3')
+        .expect(200)
+        .then((data)=>{    
+            const article = data.body
+            expect(article).toMatchObject({
+                article_id: expect.any(Number),
+                title : expect.any(String),
+                topic : expect.any(String),
+                author : expect.any(String),
+                body : expect.any(String),
+                created_at : expect.any(String),
+                votes : expect.any(Number),
+                article_img_url: expect.any(String)
+            })
+        })
```

Comment:
As it is just one object, we can assert the values as well rather than using expect.any()